File: 1161. Maximum Level Sum of a Binary Tree/1161.cs
using System;
using System.Collections.Generic;

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    private List<int> levelSums = new List<int>();

    public int MaxLevelSum(TreeNode root) {
        Dfs(root, 0);

        int maxLevel = 1; 
        int maxSum = int.MinValue;

        for (int i = 0; i < levelSums.Count; i++) {
            if (levelSums[i] > maxSum) {
                maxSum = levelSums[i];
                maxLevel = i + 1;
            }
        }

        return maxLevel;
    }

    private void Dfs(TreeNode node, int level) {
        if (node == null) return;

        if (level == levelSums.Count) {
            levelSums.Add(node.val);
        } 
        else {
            levelSums[level] += node.val;
        }

        Dfs(node.left, level + 1);
        Dfs(node.right, level + 1);
    }
}

File: 1161. Maximum Level Sum of a Binary Tree/1161.py
from collections import deque

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def maxLevelSum(self, root: [TreeNode]) -> int:

        queue = deque([root])


        big_level = 0
        big_sum = -100000

        lvl = 0
        while queue:

            sum_lvl = 0

            for _ in range(len(queue)):
                node = queue.popleft()

                sum_lvl += node.val

                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)

            if sum_lvl > big_sum:
                big_sum = sum_lvl
                big_level = lvl

            lvl += 1
        return big_level


